quickwit/quickwit-indexing/src/actors/indexing_service.rs

```diff
@@ -410,6 +392,67 @@ impl IndexingService {
         Ok(indexes_metadata)
     }
 
+    /// Fetches the immature splits candidates for merge for all the indexing pipelines for which a
+    /// merge pipeline is not running.
+    async fn fetch_immature_splits(
```

Comment:
I might be missing something: The way the code is made, I suspect we still can hammer the metastore when these index pipelines are new for instance (which is a common case).

In that case, the hashmap returned will not contain any entry for the `MergePipelineId`. 
The initialize_immature_merge will be set to `None` and as a result, we will redo the individual fetch_immature_split.

Reply:
It won't behave the way you described because of how the return `per_merge_pipeline_immature_splits` is initialized.
```
let mut per_merge_pipeline_immature_splits: HashMap<MergePipelineId, Vec<SplitMetadata>> =
            indexing_pipeline_ids
                .iter()
                .map(|indexing_pipeline_id| (indexing_pipeline_id.merge_pipeline_id(), Vec::new()))
                .collect();
```

Reply:
This is actually unit-tested.